std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -355,6 +357,53 @@ public static List<Utf16Span> span_of_all(String haystack, String needle) {
     return occurrences;
   }
 
+  /**
+   * Find spans of all occurrences of a set of needles within the haystack.
+   *
+   * @param haystack the string to search
+   * @param needles the substrings that are searched for
+   * @return a list of UTF-16 code unit spans at which the needle occurs in the haystack
+   */
+  public static List<Utf16Span> span_of_all_multiple(String haystack, List<String> needles) {
+    if (needles.isEmpty())
+      throw new IllegalArgumentException(
+              "The operation `span_of_all_multiple` does not support searching for an empty term.");
```

Comment:
I'd be tempted to do the individual needle check here as well `needles.stream().anyMatch(String::isEmpty)`.